src/resolve/resolved-dns-transaction.c

```diff
@@ -917,8 +919,16 @@ static int dns_transaction_dnssec_ready(DnsTransaction *t) {
                         if (r < 0)
                                 log_oom_debug();
 
+                        /* The answer would normally be replaced by the validated subset, but at this point
+                         * we aren't going to bother validating the rest, so just drop it. */
+                        empty = dns_answer_new(0);
+                        if (!empty)
+                                return -ENOMEM;
+                        DNS_ANSWER_REPLACE(t->answer, empty);
```

Comment:
hmm, this is unnecessarily complex. Quite widely in the systemd tree we consider NULL collections equivalent to empty collections. DnsAnswer is the same. Hence, the above four lines can be replaced by:

```c
t->answer = dns_answer_unref(t->answer);
```